marketing/tasks.py

```diff
@@ -0,0 +1,158 @@
+
+
+
+User = get_user_model()
+
+# Logging #####################################################################
+
+logger = logging.getLogger(__name__)
+
+
+# Functions ###################################################################
+
+def render_and_dispatch_email(template: EmailTemplate, application: BetaTestApplication):
+    """
+    Given an EmailTemplate and BetaTestApplication, renders the email from
+    template and sends the email.
+    """
+    context = Context({
+        "full_name": application.user.get_full_name(),
+        "username": application.user.get_username(),
+        "instance_name": application.instance.name,
+        "subdomain": application.subdomain,
+        "application": application
+    })
+    subject = template.subject_template.render(context)
+    html_body = template.html_body_template.render(context)
+    plaintext_body = template.plaintext_body_template.render(context)
+
+    try:
+        send_mail(
+            subject=subject,
+            message=plaintext_body,
+            html_message=html_body,
+            from_email=settings.DEFAULT_FROM_EMAIL,
+            recipient_list=[application.user.email],
+        )
+    except smtplib.SMTPException:
+        logger.error("Failed to send followup email '%s' to %s", template.name, application.user)
+    else:
+        SentEmail.objects.create(
+            recipient_email=application.user.email,
+            template=template,
+            subject=subject,
+            html_body=html_body,
+            plaintext_body=plaintext_body,
+        )
+
+
+def create_eligible_user_queryset():
+    """
+    Creates a queryset of eligible users for sending followup emails with
+    prefetch to reduce the number of DB calls.
+
+    Criteria for eligible user:
+        1. User.subscriber.receive_followup should be true
+        2. User.betatestapplication.instance should not be null
+    Prefetched Data:
+        1. User.betatestapplication
+        2. User.subscriber
+        3. User.betatestapplication.instance
+        4. User.sentemail_set
+        5. User.sentemail_set.template: This can be used to later filter out already sent templates.
+    """
+    eligible_users = User.objects.select_related(
+        'subscriber',
+        'betatestapplication',
+        'betatestapplication__instance',
+    ).filter(
+        subscriber__receive_followup=True, betatestapplication__instance__isnull=False
+    ).prefetch_related(
+        Prefetch(
+            'sentemail_set',
+            queryset=SentEmail.objects.all(),
+            to_attr="sent_emails"
+        ),
+        Prefetch(
+            'sentemail_set__template'
+        )
+    )
+
+# Tasks #######################################################################
+
+@db_periodic_task(crontab(day='*', hour='1', minute='0'))
+def send_followup_emails():
+    """
+    Sends all the configured followup emails.
+    """
+    emails_to_send = []
+    eligible_users = create_eligible_user_queryset()
```

Comment:
This is not efficient at all (we're potentially fetching all active users + betatest application data + instance data) all in one go.

This can be heavily simplified if we:
1. Have a timestamp of the first activated appserver in the `Subscriber` model.
2. Instead of filtering by users, first retrieve all active templates.
3. Loop over the templates, and use a queryset to filter if we're in the send range instead of doing the calculation in python. 
```
# Something along these lines
for template in active_templates:
    gte = datetime.now() - timedelta(days=template.send_after_days)
    lt = datetime.now() - timedelta(days=template.send_after_days + 1)
    to_send = Subscribers.objects.filter(created_at__gte=gte, created_at__lt=lt)
    # Prefetch stuff 
    ....
    # Render template & send
    ....
```
This way you'll only be querying for users that are actually in the send range for the emails, as opposed of retrieving all instances and doing the filtering in python.